src/utils.ts

```diff
@@ -216,3 +216,39 @@ export function arrayEquals<T>(
   if (left.length !== right.length) return false;
   return left.every((element, index) => element === right[index]);
 }
+
+/**
+ * Type that extends any non-function entity--like a string, number, or array--into a itself and a no-input function that returns it.
+ * Used to interact with objects that could either be functions or static values.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type Delayed<T> = [T] extends [(...args: any) => any]
+  ? never
+  : T | (() => T);
+
+/**
+ * Used to collapse a Delayed<T> object into an entity of type "T" as represented by the object.
+ * @param delayedObject Object of type Delayed<T> that represents either a value of type T or a function returning a value of type T.
+ * @returns The return value of the function, if delayedObject is a function. Otherwise, this returns the original element.
+ */
+export function undelay<T>(delayedObject: Delayed<T>): T {
+  return typeof delayedObject === "function" ? delayedObject() : delayedObject;
+}
+
+/**
+ * An object keyed by string type T, with values of S.
+ *  or contains a 'default' parameter to use as a fallback.
+ */
+export type Switch<T extends string, S> =
+  | Record<T, S>
+  | (Partial<{ [x in T]: S }> & { default: S });
+
+export function makeByXFunction<T extends string>(
```

Comment:
Can you document this function? What is `makeByXFunction` meant to do?

The worse a function name is the better the docs gotta be.